javascript/primitive-data-types.md

```diff
@@ -131,30 +200,288 @@ Number("100a"); // NaN
+
+Number.isFinite(1000 / 1); // true
+Number.isFinite(1000 / 0); // false
+Number.isFinite(Infinity); // false
+Number.isFinite(Number.NaN); // false
+Number.isFinite(-Infinity); // false
+Number.isFinite(null); // false
+Number.isFinite("0"); // false
+```
+
+NOTE: the difference between isFinite() and Number.isFinite(): the former will return true if the value is currently a number, OR if it is going to be after it is coerced to a number, while the latter will return true ONLY if the value is currently a number
+
+```js
+Number.isNaN() //determines whether the passed value is NaN and its type is Number
+
+⚠️ Number.isNaN() !=== window.isNaN
+
+Number.isNaN('100F'); // false
+Number.isNaN(NaN); // true
+Number.isNaN(Number.NaN); // true
+Number.isNaN(0 / 0); // true
+
+Math.sqrt(-1); // NaN
+parseInt('blabla'); // NaN
+
+// e.g. these would have been true with global isNaN()
+Number.isNaN('NaN'); // false
+Number.isNaN(undefined); // false
+Number.isNaN({}); // false
+Number.isNaN('blabla'); // false
+
+// These all return false
+Number.isNaN(true);
+Number.isNaN(null);
+Number.isNaN(37);
+Number.isNaN('37');
+Number.isNaN('');
+```
+
+NOTE: the difference between isNaN() and Number.isNaN(): the former will return true if the value is currently NaN, OR if it is going to be NaN after it is coerced to a number, while the latter will return true ONLY if the value is currently NaN
+
+```js
+isNaN("hello world"); // true
+Number.isNaN("hello world"); // false
+// however
+Number.isNaN(Number("hello world")); // true
+```
+
+```js
+Number.parseInt(value: string, radix?: number[2..36]) //parses a string argument and returns an integer of the specified radix or base.
+
+Number.parseInt('123'); // 123
+Number.parseInt('111', 2); // 7
+Number.parseInt('20', 16); // 32
+Number.parseInt('a11'); // NaN
+```
+
 ## Boolean
 
+https://developer.mozilla.org/en-US/docs/Glossary/boolean
+
 Boolean represents a logical entity and can have two values: true and false.
+Do not confuse the primitive Boolean values true and false with the true and false values of the Boolean object.
+
+Any object of which the value is not undefined or null, including a Boolean object whose value is false, evaluates to true when passed to a conditional statement. For example, the condition in the following if statement evaluates to true:
 
 ```js
+const x = new Boolean(false);
+if (x) {
+  // this code is executed as x (object) will be evaluated as true
+}
+```
 
+This behavior does not apply to Boolean primitives. For example, the condition in the following if statement evaluates to false:
+
+```js
+const x = false;
+if (x) {
+  // this code is not executed as x (boolean primitive data type) is set as false
+}
+```
+
+Creating Boolean objects with an initial value of false
+
+```js
+var bNoParam = new Boolean();
+var bZero = new Boolean(0);
+var bNull = new Boolean(null);
+var bEmptyString = new Boolean("");
+var bfalse = new Boolean(false);
+```
```

Comment:
these examples would be better using `const` and also not adding the `b` prefix but instead using more _speaking_ variables e.g.

```js
const undefinedToBoolean = new Boolean();
const zeroToBoolean = new Boolean(0);
const nullToBoolean = new Boolean(null);
const emptyString = new Boolean("");
const falseValue = new Boolean(false);
```

also what do they evaluate to?